[RyuJIT/arm32][JitStressRegs=8] Assertion failed 'this->gtOper == GT_LIST || this->gtOper == GT_FIELD_LIST'

From corefx x86_arm_altjit run, e.g. "@dotnet-bot test Windows_NT x86_arm_altjit Checked corefx_jitstressregs8".

Happens with `COMPlus_JitStressRegs=8`.

Found in System.Diagnostics.FileVersionInfo.Tests:
```
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>RunTests.cmd E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\
Using E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\ as the test runtime folder.
Executing in E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86\
Running tests... Start time: 14:44:34.59

E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJit=*
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitNgen=*
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitName=protononjit.dll
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_NoGuiOnAssert=1
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_ContinueOnAssert=0
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitAssertOnNYI=1
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_JitStressRegs=8

E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>call E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\\dotnet.exe xunit.console.netcore.exe System.Diagnostics.FileVersionInfo.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonwindowstests  -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Diagnostics.FileVersionInfo.Tests
Discovered:  System.Diagnostics.FileVersionInfo.Tests
Starting:    System.Diagnostics.FileVersionInfo.Tests

Assert failure(PID 23496 [0x00005bc8], Thread: 10464 [0x28e0]): Assertion failed 'this->gtOper == GT_LIST || this->gtOper == GT_FIELD_LIST' in 'System.Diagnostics.FileVersionInfo:GetVersionInfoForCodePage(int,ref):bool:this' (IL size 612)

    File: e:\gh\coreclr2\src\jit\gtstructs.h Line: 68
    Image: E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\dotnet.exe
```

Full repro:
```
dotnet/coreclr:
E:\gh\coreclr2> build.cmd x86 checked

dotnet/corefx:
E:\gh\corefx> build.cmd -Release -os:Windows_NT -buildArch:x86 -- /p:CoreCLROverridePath=E:\gh\coreclr2\bin\Product\Windows_NT.x86.Checked
E:\gh\corefx> build-tests.cmd -Release -os:Windows_NT -buildArch:x86 -- /p:WithoutCategories=IgnoreForCI /p:PreExecutionTestScript=e:\bugs\SetStressModes.bat

With e:\bugs\SetStressModes.bat:
set COMPlus_AltJit=*
set COMPlus_AltJitNgen=*
set COMPlus_AltJitName=protononjit.dll
set COMPlus_NoGuiOnAssert=1
set COMPlus_ContinueOnAssert=0
set COMPlus_AltJitAssertOnNYI=1
set COMPlus_JitStressRegs=8
```

```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  cat tmp/coreoverlay-ARM-CHE-REL-10181724/COMMITS.CLR 
e869941a7364ef128a50ecdfce0a9a0e317d13c2 format spmi sources. (#14545)
a194ac80eb0e97e822b9eb35c24c59c3c24d7693 Merge pull request #14553 from briansull/fix-warn
882b1d64a179f2d04dc66c8845b723f06c30bdea Merge pull request #14555 from wtgodbe/AzureBlobSymPkg
de4559f6dbaa7fd732f6529e93b0aebde0fb0ea9 Use C#-friendly path for indexing symbol packages
d183b9f0cb9235440902eb6f80841830d96f5775 Fix for x86 desktop build Fix warning C4389: '==' : signed/unsigned mismatch
4fa06e6095af6473d2ec6c855ebfb8d77017d91d Merge pull request #14541 from stephentoub/taskbeginwait
2f01c285d1a2d01cf9d8810e557961872c1eb46d GetEnvironmentVariable: Avoid StringBuilder marshaling overhead (#14502)
2cd4ca4cdd8b71563a58835850aa67de5906821c CLRLifoSemaphore cleanup (#14535)
0526339a302da4918699901d5fc403044374651b Avoid TaskWaitBegin/End events in Task.Wait on already completed task
b5616f0013ccba23c6a3e6facbb09545e710a5f1 Merge pull request #14493 from alpencolt/armel-cross-build
```

```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  cat tmp/coreoverlay-ARM-CHE-REL-10181724/COMMITS.FX 
5c8af507d63e94a49ab4b4a81a45510a50146faf Adding ValidateTypeIsBlittable SpanHelper method and rename constant (#24626)
b28112eb892283f20e7e422a52da0a55ca4881d5 Rewrite BackgroundWorker test to be non blocking (#24693)
ae86eca620025087b576d092e67acde7760d4baa Deleted unused string resources. (#24697)
a1e7009bd3c94c3999bfd471a08e5d647c571e52 Add null check when setting Packaging property (#24673)
0ad10b8bf74a4ce2fe452aa23801b3722237560d Update CoreSetup, Standard to preview1-25817-01, preview1-25817-02, respectively (#24702)
5ef5f17b5606377415d5c171d6613c49c59a1b55 Set.TryGetValue not-found behavior/docs inconsistent (#24699)
79089260b0bf00e99ba9dd7e736c7b2d776fed0a Deleted dead code from System.IO.FileSystem. (#24701)
3b867880d9992f305757db85a9b32044f9773863 Fix for GraphicsPathIterator.CopyData throws EntrypointNotFoundException when index is out of range. (#24656)
e3977683cf78b7b1896a11e9de7df6a62408a1a6 Remove methods marked as internal from reference assembly. (#24700)
db596d27932201ec4936a7c9baa6aa6ad3f9e796 Update XunitNetcoreExtensionsVersion to 1.0.1-prerelease-02116-01 (#24687)
```

--- a/src/jit/gentree.cpp
+++ b/src/jit/gentree.cpp
@@ -5561,11 +5561,12 @@ bool GenTree::TryGetUse(GenTree* def, GenTree*** use)
 
         case GT_FIELD_LIST:
             return TryGetUseList(def, use);
+
 #if !defined(LEGACY_BACKEND) && defined(_TARGET_ARM_)
         case GT_PUTARG_SPLIT:
             if (this->AsUnOp()->gtOp1->gtOper == GT_FIELD_LIST)
             {
-                return TryGetUseList(def, use);
+                return this->AsUnOp()->gtOp1->TryGetUseList(def, use);
             }
             if (def == this->AsUnOp()->gtOp1)
             {

